index.js

```diff
@@ -1,39 +1,86 @@
-const express = require('express')
-const app = express()
-const port = 3001
-
-// app.get('/', (req, res) => res.send('Steam reviews service?'))
+const db = require('./db/connection.js');
+const express = require('express');
+const app = express();
+const port = 3001;
 
 app.listen(port, () => console.log(`Steam reviews service. listening at http://localhost:${port}`))
 app.use(express.static('./client/dist'));
 
-// SAMPLE API CALLS
 
-app.get('/api/reviewcount/all/summary/:productId', ( req, res) => {
-    res.send( sampleAllSummary );
-});
+// tool: return summary text from %
+var rating = per => 
+[ { min: 95, text:'Overhwelmingly Positive' },
+  { min: 80, text:'Very Positive' },
+  { min: 70, text:'Mostly Positive' },
+  { min: 40, text:'Mixed' },
+  { min: 20, text:'Mostly Negative' },
+  { min: 0, text:'Very Negative' } ]
+  .find( row => per >= row['min'] )['text'];
 
-app.get('/api/reviewcount/all/:productId', ( req, res) => {
-  res.send( sampleAll );
-});
 
-app.get('/api/reviewcount/recent/summary/:productId', ( req, res) => {
-  res.send( sampleRecentSummary );
+app.get('/api/reviewcount/:gameId', ( req, res ) => {
+
+  var sqlText = `SELECT SUM(positive) as pos, SUM(negative) as neg ` +
+                `FROM reviews_graph WHERE gameid = ${ req.params.gameId };`;
+
+  db.query( sqlText, ( err, result, fields ) => {
+      if (err) throw err;
```

Comment:
In addition to or instead of throwing the error, maybe `res.status(500).json({ error: 'Internal server error' })` would be good, so that the request isn't left hanging (since throwing would terminate the `app.get` call without a `res.send()`/`res.json()`/`res.end()` of some kind)

Reply:
(now) calling generic error to the client then throwing internal error.